graphql/provider.go

```diff
@@ -40,10 +68,67 @@ func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{},
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
+
+	if d.Get("login_query") != "" {
+		queryResponse, resBytes, err := queryExecute(ctx, d, config, "login_query", "login_query_variables")
+		if err != nil {
+			return nil, diag.FromErr(fmt.Errorf("unable to execute read query: %w", err))
+		}
+
+		if queryErrors := queryResponse.ProcessErrors(); queryErrors.HasError() {
+			return nil, *queryErrors
+		}
+
+		var queryResponseCty cty.Value
+		if queryResponseCty, err = gocty.ToCtyValue(string(resBytes), cty.String); err != nil {
+			return nil, diag.FromErr(err)
+		}
+
+		evalCtx := &hcl.EvalContext{
+			Variables: map[string]cty.Value{
+				"login_query_response": queryResponseCty,
+			},
+			Functions: map[string]function.Function{
+				"jsondecode": stdlib.JSONDecodeFunc,
+				//TODO: maybe add other useful functions
```

Comment:
I currently only included support for the `jsondecode()`  function as it's the only one I needed in my use case.

But we may want to add others.

For reference: 
- https://github.com/hashicorp/hcl/blob/v2.11.1/cmd/hcldec/spec_funcs.go#L8
- https://github.com/hashicorp/terraform/blob/v1.1.2/internal/lang/functions.go#L34

Reply:
This is interesting. I believe JSONdecode is the sane default to use here. I am not very familiar with this pattern, so I need to read the docs on this to gain a better understanding of what is happening here. If my understanding is correct, I imagine jsondecode should suffice.